test/functional/homepage_controller_test.rb

```diff
@@ -17,4 +17,103 @@ class HomepageControllerTest < ActionController::TestCase
       assert_template "homepage/popular_links/show"
     end
   end
+
+  context "#create" do
+    setup do
+      @popular_links = create_a_published_popular_links
+    end
+    should "render show template" do
+      post :create, params: { id: @popular_links.id }
+
+      assert_response :ok
+      assert_template "homepage/popular_links/show"
+    end
+
+    should "create a new draft popular links" do
+      assert_equal 1, PopularLinksEdition.count
+      assert_equal "published", PopularLinksEdition.last.state
```

Comment:
Do we need these assertions at the start? Aren't they just testing that the setup code worked?

Similar comment for subsequent tests.